### PR TITLE
enable attachments for symbols. fixes SI-5975, an ide bug with default args.

### PR DESCRIPTION
**Pull request by lrytz**
gets rid of the global hash maps in Namers and NamesDefaults

review by @dragos. i assume there should not be memory leaks, the symbol attachments should get GC'd together with the symbol

review by @dotta to make sure i merge-squashed his code correctly
review by @xeno-by for symbol attachments

### CODE REVIEW

**Comment by lrytz** [comment]
PLS REBUILD ALL

**Comment by dotta** [comment]
lgtm. Thanks Lukas, amazing job!

**Comment by xeno-by** [comment]
1) We have stdattachments.scala that store compiler-wide attachment
case classes. Let's use them to be uniform.
2) Why copypaste stuff from Trees into Symbols instead of inheriting
both from a trait Attachable?
3) Attachments need to be exposed in the makro context API,
analogously to tree attachments.

On 4 Jul 2012, at 14:17, Mirco Dotta
reply@reply.github.com
wrote:

**Comment by lrytz** [comment]
1) yeah, all the same for me. i found also a few attachment classes in other places (eg `DefaultOverrideMatchAttachment`). i thought it's better to put them in context and document what they are used for.

2) can do that

3) i don't see where tree attachments are in the macro api. as far as i can see, tree attachments exist only in `reflect/internal/Trees.scala`

**Comment by xeno-by** [comment]
1) @adriaanm, what do you think about this?

3)
https://github.com/scalamacros/kepler/blob/691aa0981e701c73df01ad9d5c6dd8aafb289980/src/reflect/scala/reflect/makro/Universe.scala#L67

On 4 July 2012 18:01, Lukas Rytz <
reply@reply.github.com

**Comment by adriaanm** [comment]
1) I'm not sure. Intuitively, or maybe it's just inertia, I'd expect the attachments to be defined where they're most relevant.
I can't think of a reason why you'd need a list of all possible attachments. If there is one, that would be a convincing argument to keep them in one place.

**Comment by adriaanm** [comment]
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/view/pr-validators/job/pr-scala-testsuite-linux-opt/411/

**Comment by adriaanm** [comment]
```
  [partest] testing: [...]/files/presentation/ide-t1000976                        [FAILED]
  [partest] not found: object c
  [partest] not found: type C
```

**Comment by lrytz** [comment]
seems the ide testing infrastructure still has problems with sourcepath. either that or i messes up, i'll check.

i close this one and submit a new one with the discussed changes.

File: test/files/presentation/ide-t1000976.flags [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-sourcepath files/presentation/ide-t1000976/src
```
Comment (dotta) [inline test/files/presentation/ide-t1000976.flags:1]: Oh, yeah, this is wrong, it should be:

`-sourcepath src`

Looks like you missed [this commit](https://github.com/dotta/scala/commit/bf3cff6f2b2115944a2bb6f72f96193e58a9508e) when merging my branch.

How did I miss it :)

Anyway, if you merge that commit,  then it works fine.

**Comment by lrytz** [comment]
Sorry, Mirco. Thanks for tracking it down.

**Pull request closed without merging**